cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/datastore/DatastoreManager.java

```diff
@@ -333,7 +333,7 @@ private Datastore createDatastore(String dbName, KeyProvider provider) throws Da
             // if it does not exist
 
             //Pass database directory, database name, and SQLCipher key provider
-            DatastoreImpl ds = new DatastoreImpl(dbDirectory, dbName, provider);
+            QueryableDatastore ds = new QueryableDatastore(dbDirectory, dbName, provider);
```

Comment:
Having the `DatastoreImpl` delegate to `Queryable` would cause abstraction problems. Since we'd have to decouple them and `Queryable` relies on Datastore APIs, which is why I selected a subclass. 

w.r.t. start up time, the update schema methods are async, so shouldn't be an issue. FTS is sync, however but if that goes down onto the `SQLDatabaseQueue` we could push the check to when we access the property for the first time. Otherwise there is no real work occurring in the constructor, e.g. we don't update indexes or anything.